src/main/java/org/eclipse/tractusx/managedidentitywallets/domain/Identifier.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * *******************************************************************************
+ *  Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ * ******************************************************************************
+ */
+
+package org.eclipse.tractusx.managedidentitywallets.domain;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Identifier is a unique identifier for a wallet.
+ */
+public class Identifier {
+    private static final String PATTERN_DID = "^did:web:([a-z0-9\\\\.]*\\b(%3A\\d{2,5})?\\b)\\b:BPN[ALS][0-9a-f]{12}$";
+
+    private static final String PATTERN_BPN = "^BPN[ALS][0-9a-f]{12}$";
+
+    private final String value;
+
+    private final boolean isDid;
+
+    public Identifier(final String identifier) {
+
+        Pattern didPattern = Pattern.compile(PATTERN_DID);
+        Matcher didMatcher = didPattern.matcher(identifier);
+        boolean didMatched = didMatcher.matches();
+
+        if(!didMatched){
+            Pattern bpnPattern = Pattern.compile(PATTERN_BPN);
+            Matcher bpnMatcher = bpnPattern.matcher(identifier);
+            if(!bpnMatcher.matches()){
+                throw new IllegalArgumentException("identifier %s is not valid".formatted(identifier));
```

Comment:
I find that creating our own excpetion handling is important.

we can create in the future our own handling mechanism, the main goal of this refactor wasn't to also create a error hanlding mechanism or a logging system as these mechansims should be first planned together and discussed.

we used IllegalArgumentException because of this https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html
the summary of that link is this " Thrown to indicate that a method has been passed an illegal or inappropriate argument. "

in this case springboot will throw 400 Error with a message that looks like this 
 "IllegalArgumentException: identifier did:web:localhost:BPNLF00000000000 is not valid"

Springboot keep tracking of what the client as a request param gave and if there is a validation error with it or not, as in the case of creating a BPN or Identifier out of client input then it will throws 400 error which indicates a client side error.

I understand that there is a big room into creating even more features and improved and i beleive that we will do that step for step together!

if you still have any concerns about this now we can still discuss that.

Reply:
We have BadDataException already defined in MIW. can we use it?